File: pickerview/src/main/java/org/jaaksi/pickerview/adapter/NumericWheelAdapter.java
package org.jaaksi.pickerview.adapter;

/**
 * 数字adapter
 */
public class NumericWheelAdapter implements WheelAdapter<Integer> {

  private int minValue;
  private int maxValue;

  /**
   * Constructor
   *
   * @param minValue the wheel min value
   * @param maxValue the wheel max value
   */
  public NumericWheelAdapter(int minValue, int maxValue) {
    this.minValue = minValue;
    this.maxValue = maxValue;
  }

  @Override public int getItemCount() {
    return maxValue - minValue + 1;
  }

  @Override public Integer getItem(int index) {
    if (index >= 0 && index < getItemCount()) {
      return minValue + index;
    }
    return 0;
  }
}

File: pickerview/src/main/java/org/jaaksi/pickerview/adapter/NumericWheelAdapter.kt
package org.jaaksi.pickerview.adapter

/**
 * 数字adapter
 *
 * @param minValue the wheel min value
 * @param maxValue the wheel max value
 */
class NumericWheelAdapter(private val minValue: Int, private val maxValue: Int) :
    WheelAdapter<Int> {
    override val itemCount: Int
        get() = maxValue - minValue + 1

    override fun getItem(index: Int): Int {
        return if (index in 0 until itemCount) minValue + index else 0
    }
}